Update Billing.md (#3584)
elastic/integrations: packages/aws/_dev/build/docs/billing.md

# Billing

## Overview

The [AWS Billing Console](https://aws.amazon.com/aws-cost-management/aws-billing/) integration allows you to monitor you AWS Billing usage.

> Use the AWS Billing Console integration to collect and parse logs related to your monthly AWS bills. Then visualize that data in Kibana, create alerts to notify you if something goes wrong, and reference logs when troubleshooting an issue.

Reviewer: A couple notes here:
* Use the proper name of the integration, which is "AWS Billing integration".
* The AWS Billing integration does not collect _logs_, it collects _metrics_. Logs and metrics are two distinct types of data streams. If you look at [the original doc](https://docs.elastic.co/integrations/aws/billing) for this integration you can see that it references _metrics_ not _logs_. (Some integrations have one or the other or both.)

```suggestion
Use the AWS Billing integration to collect metrics related to your monthly AWS bills. Then visualize that data in Kibana, create alerts to notify you if something goes wrong, and reference metrics when troubleshooting an issue.
```

> Use the AWS Billing Console integration to collect and parse logs related to your monthly AWS bills. Then visualize that data in Kibana, create alerts to notify you if something goes wrong, and reference logs when troubleshooting an issue.

Reviewer 2: We are not collecting any logs to parse billing here.

Author: I've added details on the metrics collected

For example, you could use this data to view your your monthly chargeable costs for each AWS account you have. Then you can alert the relevant budget holder about those costs by email.

## Data streams

The AWS Billing Console integration collects one type of data: logs.

> **Logs** help you keep a record of events happening in Billing.

Reviewer 2: should be metrics

Author: changed by Colleen

> ## Data streams
> 
> The AWS Billing Console integration collects one type of data: logs.
> 
> **Logs** help you keep a record of events happening in Billing.
> Log data streams collected by the AWS Billing Console integration include your monthly chargeable costs, along with details of your AWS services and purchases made through AWS Marketplace, and more. See more details in the [Logs](#logs-reference).

Reviewer: As mentioned above, this integration collects metrics not logs. Use the [new guidelines](https://github.com/colleenmcginnis/integrations/blob/top-five-audit-guidelines/docs/documentation_guidelines.md#data-streams) as a template for drafting this section. The suggestion below deviates a bit from the template language because there is only one metric data stream.

>include your monthly chargeable costs, along with details of your AWS services and purchases made through AWS Marketplace, and more

I'm not sure if this is true. Looking at the [example event in the existing doc](https://docs.elastic.co/integrations/aws/billing#metrics), it looks like it provides estimated costs per service for a given time period.

```suggestion
## Data streams

The AWS Billing Console integration collects one type of data: metrics.

**Metrics** give you insight into the state of your AWS bills including the estimated costs for various AWS services.
See more details in the [Metrics](#metrics-reference).
```

## Requirements

You need Elasticsearch for storing and searching your data and Kibana for visualizing and managing it.
You can use our hosted Elasticsearch Service on Elastic Cloud, which is recommended, or self-manage the Elastic Stack on your own hardware.

 Before using any AWS integration you will need:

 * **AWS Credentials** to connect with your AWS account.
 * **AWS Permissions** to make sure the user you're using to connect has permission to share the relevant data.

 For more details about these requirements, see the **AWS** integration documentation.

## Setup

> Use this integration if you only need to collect data from the AWS Billing service.

Reviewer: ```suggestion
Use this integration if you only need to collect billing data from the AWS.
```

If you want to collect data from two or more AWS services, consider using the **AWS** integration. When you configure the AWS integration, you can collect data from as many AWS services as you'd like.

For step-by-step instructions on how to set up an integration, see the
[Getting started](https://www.elastic.co/guide/en/welcome-to-elastic/current/getting-started-observability.html) guide.

> ## Logs reference
> 
> The `billing` data stream collects standard logs (also called access logs) from AWS Billing. Billing standard logs provide detailed records about every request that’s made to the AWS Billing console.

Reviewer: ```suggestion
## Metrics reference

The `billing` data stream collects billing metrics from AWS.

An example event for `billing` looks as following:
```

# AWS Billing

> ## Overview

Reviewer: I think that's a good idea.

> ## Overview

Author: I've removed the heading

The AWS Billing integration allows you to monitor your [AWS costs](https://aws.amazon.com/aws-cost-management/aws-billing/).

Use the AWS Billing integration to collect metrics related to your monthly AWS bills. Then visualize that data in Kibana, create alerts to notify you if something goes wrong, and reference metrics when troubleshooting an issue.

For example, you could use this data to view your your monthly chargeable costs for each AWS account you have. Then you can alert the relevant budget holder about those costs by email.

## Data streams

The AWS Billing integration collects one type of data: metrics.

> **Metrics** give you insight into the state of your AWS bills including the estimated costs for various AWS services ([Cloudwatch](https://aws.amazon.com/cloudwatch/) and AWS [Cost Explorer API](https://docs.aws.amazon.com/cost-management/latest/userguide/ce-api.html)).

Reviewer 3: Where does this Cloudwatch and Cost Explorer API information come from? I can't find it mentioned in the integration.

Author: From [kaiyan-sheng](https://github.com/kaiyan-sheng)'s review earlier:

_We are not collecting any logs to parse billing here. Billing metrics from two sources: cloudwatch and AWS [cost explorer API](https://docs.aws.amazon.com/cost-management/latest/userguide/ce-api.html)._

Reviewer 2: Sorry it's documented in Beats repo https://github.com/elastic/beats/blob/main/x-pack/metricbeat/module/aws/billing/_meta/docs.asciidoc

Reviewer 3: Neat, thanks!

See more details in the [Metrics](#metrics-reference).

## Requirements

You need Elasticsearch for storing and searching your data and Kibana for visualizing and managing it.
You can use our hosted Elasticsearch Service on Elastic Cloud, which is recommended, or self-manage the Elastic Stack on your own hardware.

 Before using any AWS integration you will need:

 * **AWS Credentials** to connect with your AWS account.
 * **AWS Permissions** to make sure the user you're using to connect has permission to share the relevant data.

>  For more details about these requirements, see the **AWS** integration documentation.

Reviewer 3: It really is a shame we can't turn this into a link. Nothing for you to do here, @ellis-elastic.

Reviewer 3: Or, maybe we could. @colleenmcginnis should this be a link to our public docs for the AWS integration?

Reviewer: From https://github.com/elastic/integrations/pull/3308#pullrequestreview-968332089:

>Hmm. I don't think this link will work in either Kibana or the doc build. We either need to remove this, or hard link to https://docs.elastic.co/integrations/aws/cloudfront.
>
>Just tested this and my suspicion is correct. All of these relative links will need to be updated 😢

In that PR I just removed all links to other integration pages, but we could add absolute links. It sounds like that would just link out from Kibana to the docs if they are viewing info about the integration in Kibana, but that might be ok!

Reviewer 3: Whatever you think is better. I'm fine leaving this as-is for now.
